src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/Utils.scala

```diff
@@ -55,4 +62,80 @@ object Utils {
       tenant.attributes.mapFilter(_.certified.map(_.id)) ++
       tenant.attributes.mapFilter(_.declared.map(_.id))
 
+  def isUpgradable(
+    descriptor: CatalogManagement.EServiceDescriptor,
+    descriptors: Seq[CatalogManagement.EServiceDescriptor]
+  ): Boolean =
+    descriptors
+      .filter(_.version.toInt > descriptor.version.toInt)
+      .exists(d =>
+        d.state == CatalogManagement.EServiceDescriptorState.PUBLISHED ||
+          d.state == CatalogManagement.EServiceDescriptorState.SUSPENDED
+      )
+
+  def isUpgradable(
+    descriptor: CatalogProcess.EServiceDescriptor,
+    descriptors: Seq[CatalogProcess.EServiceDescriptor]
+  ): Boolean =
+    descriptors
+      .filter(_.version.toInt > descriptor.version.toInt)
+      .exists(d =>
+        d.state == CatalogProcess.EServiceDescriptorState.PUBLISHED ||
+          d.state == CatalogProcess.EServiceDescriptorState.SUSPENDED
+      )
+
+  def getAllAgreements(
+    producersIds: List[UUID],
+    consumersIds: List[UUID],
+    eServicesIds: List[UUID],
+    states: List[AgreementProcess.AgreementState],
+    ec: ExecutionContext,
+    agreementProcessService: AgreementProcessService
```

Comment:
In relation to [this](https://github.com/pagopa/interop-be-backend-for-frontend/pull/137/files#r1126835266), passing the AgreementProcessService would become superfluous.